src/scripts.js

```diff
@@ -89,24 +108,39 @@ function toggleHidden(element) {
 }
 
 function collapseRecipe(event) {
-  if(event.target.id === "recipeModalBackground"){
+  savedView ? createRecipeCards(currentUser.savedRecipes) : createRecipeCards(recipeRepo.recipes);
+  if (event.target.id === "recipeModalBackground"){
     toggleHidden(modalSection);
-  }
+  } 
 }
 
 function filterRecipes(event) {
     let tag = event.target.innerText.toLowerCase();
-    let filteredRecipes = recipeRepo.filterByTag(tag);
+    let filteredRecipes = savedView ? currentUser.filterSavedByTag(tag) : recipeRepo.filterByTag(tag);
     createRecipeCards(filteredRecipes);
 }
 
 function searchRecipes() {
   let keyword = searchBar.value;
-  let searchedRecipes = recipeRepo.filterByName(keyword);
+  let searchedRecipes = savedView ? currentUser.filterSavedByName(keyword) : recipeRepo.filterByName(keyword);
   createRecipeCards(searchedRecipes);
 }
 
 function toggleSaveRecipe(event) {
-console.log(modalRecipe)
-    // currentUser.toggleSaveRecipe(modalRecipe)
+  currentUser.toggleSaveRecipe(modalRecipe);
+}
+
+function getRandomUser(){
+  currentUser = new User(usersData[Math.floor(Math.random() * usersData.length)]);
+  navUserInfo.innerText = currentUser.name;
+}
+
+function showSavedRecipes() {
+  savedView = true;
+  createRecipeCards(currentUser.savedRecipes);
+}
+
+function goHome() {
```

Comment:
We might want to consider adding an All Recipes button to Nav that calls this function